File: Mod3/Clase09/app/src/main/java/com/area51/clase09/database/MetodoSQL.java
package com.area51.clase09.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.area51.clase09.modelos.Ubicacion;

import java.util.ArrayList;

public class MetodoSQL {
    private SQLiteManage sqLiteManage;

    public MetodoSQL(Context context) {
        sqLiteManage = new SQLiteManage(context);
    }

    public boolean agregar(String direccion,
                           Double latitud,
                           Double longitud) {
        SQLiteDatabase db = sqLiteManage.getWritableDatabase();

        try {
            db.beginTransaction();

            ContentValues values = new ContentValues();
            values.put("direccion", direccion);
            values.put("latitud", latitud);
            values.put("longitud", longitud);
            db.insert("ubicaciones", null, values);

            db.setTransactionSuccessful();
            return true;
        } catch (Exception e) {

        }
        finally {
            db.endTransaction();
        }
        return false;
    }

    public ArrayList<Ubicacion> obtener() {
        SQLiteDatabase db = sqLiteManage.getReadableDatabase();
        Cursor cursor = db.rawQuery(
                "select * from ubicaciones", null
        );
        ArrayList<Ubicacion> lista = new ArrayList<>();
        if (cursor.moveToFirst()) {
            do {
                Ubicacion ubicacion = new Ubicacion();
                ubicacion.setId(cursor.getInt(cursor.getColumnIndex("id")));
                ubicacion.setDireccion(cursor.getString(cursor.getColumnIndex("direccion")));
                ubicacion.setLatitud(cursor.getDouble(cursor.getColumnIndex("latitud")));
                ubicacion.setLongitud(cursor.getDouble(cursor.getColumnIndex("longitud")));

                lista.add(ubicacion);
            } while (cursor.moveToNext());
        }
        return lista;
    }
}

File: Mod3/Clase09_1/app/src/main/java/com/area51/clase09_1/database/MetodoSQL.kt
package com.area51.clase09_1.database

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import com.area51.clase09_1.modelos.Ubicacion
import java.lang.Exception

class MetodoSQL(context: Context) {
    var database: SQLiteManage? = null

    init {
        database = SQLiteManage(context)
    }

    fun agregar(
        direccion: String,
        latitud: Double,
        longitud: Double
    ): Boolean {
        val db = database!!.writableDatabase
        try {
            db.beginTransaction()

            val values = ContentValues()
            values.put("direccion", direccion)
            values.put("latitud", latitud)
            values.put("longitud", longitud)
            db.insert("ubicaciones", null, values)

            db.setTransactionSuccessful()
            return true
        } catch (e: Exception) {

        } finally {
            db.endTransaction()
        }
        return false
    }

    fun obtener(): ArrayList<Ubicacion> {
        val db = database!!.readableDatabase
        val cursor = db.rawQuery(
            "select * from ubicaciones", null
        )
        val lista = ArrayList<Ubicacion>()
        if (cursor.moveToFirst()) {
            do {
                val ubicacion = Ubicacion()
                ubicacion.id = cursor.getInt(cursor.getColumnIndex("id"))
                ubicacion.direccion = cursor.getString(cursor.getColumnIndex("direccion"))
                ubicacion.latitud = cursor.getDouble(cursor.getColumnIndex("latitud"))
                ubicacion.longitud = cursor.getDouble(cursor.getColumnIndex("longitud"))
                lista.add(ubicacion)
            } while (cursor.moveToNext())
        }
        return lista
    }
}